### PR TITLE
Bump Electron to version 26.2.4

### LINKED ISSUE #5619: Update Electron
After version 20.3.8, a sandbox was added to electron so that anything that tries to read from memory using pointers (i.e. `ref-napi` that has been used by `ffi-napi`) can crash electron [^1]. In addition to resolving all other issues that might come with an electron upgrade, we will also need to solve the sandboxing problem. Thankfully, `win-ca.ts` is the only identified area that is causing issues at this point.

## Proposed solution

- Utilize TLS module (short-term solution)
- Rewrite `win-ca.ts` in Go? (#5637 long-term solution)

## Acceptance Criteria

- [ ] Thoroughly tested on Windows, macOS, & Linux to ensure Rancher Desktop functions correctly with the updated Electron version
- [ ] Verify that all core functionality remains intact
- [ ] Regression tested to identify and fix any new issues introduced

[^1]: https://www.electronjs.org/blog/v8-memory-cage

### LINKED ISSUE #5623: Upgrade nodejs to v18.16
v18.16 is supported by our target electron version, 26.0.0

### PR DESCRIPTION
**Pull request by rak-phillip**
This bumps electron to 26.2.4 and node to version 18.16.x (the supported version of node for the 26.x release of electron [^1])

In order to keep this change primarily focused on making sure that we can get Rancher Desktop running with a supported & stable version of Electron, we will need to follow up with additional PRs to complete the transition:

- #5637
- #5659

closes #5619 
closes #5623

[^1]: https://www.electronjs.org/docs/latest/tutorial/electron-timelines#timeline

### CODE REVIEW

**Comment by rak-phillip** [comment]
We won't merge this until after the 1.11 release, but I think that this PR is in a state that is ready to review. Please note that the required node version has been updated to 18.16.x and we might need to reinstall yarn depending on how you manage your node versions.

File: package.json
Comment (mook-as) [inline package.json:11]: Please update `README.md` to say this node version. (There are multiple places within that file.)

File: pkg/rancher-desktop/utils/protocols.ts
Comment (mook-as) [inline pkg/rancher-desktop/utils/protocols.ts:80]: Please add a comment on why this is needed. You already have the follow-up filed as #5659 (thanks!), so maybe mention that too.

File: pkg/rancher-desktop/layouts/dialog.vue
Comment (mook-as) [inline pkg/rancher-desktop/layouts/dialog.vue:72]: It would be good to have more detail in the commit message about what this is fixing (that is, why this works).

File: pkg/rancher-desktop/window/index.ts
Comment (mook-as) [inline pkg/rancher-desktop/window/index.ts:1]: Again, more details in the commit message on what we're doing would help.

File: e2e/credentials-server.e2e.spec.ts
Comment (mook-as) [inline e2e/credentials-server.e2e.spec.ts:286]: The commit message could mention things about not supporting IPv6.

**Review by mook-as — CHANGES_REQUESTED** [review]
Looks mostly fine, just some details and changes to commit messages.

**Review by mook-as — APPROVED** [review]
Seems to run (on Windows) and can still build the MSI installer.

**Comment by rak-phillip** [comment]
@mook-as rebased on top of main. Let me know if your approval still stands. If so, I intend to merge as soon as is conveniently possible 🙂

**Pull request merged**
